src/js_collection.hpp

```diff
@@ -48,8 +48,14 @@ typename T::Value CollectionClass<T>::create_collection_change_set(ContextType c
 {
     ObjectType object = Object::create_empty(ctx);
     std::vector<ValueType> deletions, insertions, modifications;
-    for (auto index : change_set.deletions.as_indexes()) {
-        deletions.push_back(Value::from_number(ctx, index));
+    
+    if (change_set.deletions.count() == std::numeric_limits<size_t>::max()) {
+        deletions.push_back(Value::from_undefined(ctx));
```

Comment:
`undefined` is usually developer "written code". I have a **strong** opinion that libraries or frameworks should never return to the caller anything that is `undefined` unless it's introspecting metadata of some sort. 

1. `undefined` in JavaScript means that the variable has been declared but not assigned anything.
2. `null` means a representation of no value.